ui/src/app/pages/signed-in/signed-in.tsx

```diff
@@ -171,7 +170,6 @@ export const SignedInImpl = (spinnerProps: WithSpinnerOverlayProps) => {
           )}
       </FlexRow>
       {!hideFooter && <Footer type={FooterTypeEnum.Workbench} />}
-      <InactivityMonitor />
```

Comment:
The fix: The `InactivityMonitor` component was only rendered on "signed-in" pages as a proxy for whether the user was authenticated or not. If the user visited a page such as `/login` the inactivity cookie did not reset. When the user signed in again, `InactivityMonitor` treated this as visiting the page for the first time since the cookie was set. Now, the cookie is reset if the user is unauthenticated at any point, which is often the case on the `/login` page.

Overall, this fix feels like a band-aid to me rather than a full logical fix. However, the inactivity logic is surprisingly complex at the moment. More work is needed to simplify the inactivity checker.